File: bindings/lua/Polycode/CoreMotionEvent.lua
require "Polycode/Event"

class "CoreMotionEvent" (Event)


function CoreMotionEvent:__getvar(name)
	if name == "amount" then
		local retVal = Polycode.CoreMotionEvent_get_amount(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
	if Event["__getvar"] ~= nil then
		return Event.__getvar(self, name)
	end
end

function CoreMotionEvent:__setvar(name,value)
	if name == "amount" then
		Polycode.CoreMotionEvent_set_amount(self.__ptr, value.__ptr)
		return true
	end
	if Event["__setvar"] ~= nil then
		return Event.__setvar(self, name, value)
	else
		return false
	end
end
function CoreMotionEvent:CoreMotionEvent(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "Event" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.CoreMotionEvent(unpack(arg))
	end
end

function CoreMotionEvent:__delete()
	if self then Polycode.delete_CoreMotionEvent(self.__ptr) end
end

File: bindings/javascript/Polycode/CoreMotionEvent.js
require('Polycode/Event')

function CoreMotionEvent() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.CoreMotionEvent()
	}
	Object.defineProperties(this, {
		'amount': { enumerable: true, configurable: true, get: CoreMotionEvent.prototype.__get_amount, set: CoreMotionEvent.prototype.__set_amount}
	})
}


CoreMotionEvent.prototype = Object.create(Event.prototype)

CoreMotionEvent.prototype.__get_amount = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.CoreMotionEvent__get_amount(this.__ptr)
	return retVal
}

CoreMotionEvent.prototype.__set_amount = function(val) {
	Polycode.CoreMotionEvent__set_amount(this.__ptr, val.__ptr)
}

Duktape.fin(CoreMotionEvent.prototype, function (x) {
	if (x === CoreMotionEvent.prototype) {
		return;
	}
	Polycode.CoreMotionEvent__delete(x.__ptr)
})
